src/Cli/dotnet/ToolPackage/ToolPackageInstance.cs

```diff
@@ -97,13 +97,21 @@ public ToolPackageInstance(PackageId id,
             ResolvedPackageVersion = Version;
         }
 
-        var toolConfiguration = DeserializeToolConfiguration(library, packageDirectory, _fileSystem);
-        Warnings = toolConfiguration.Warnings;
-
         var installPath = new VersionFolderPathResolver(PackageDirectory.Value).GetInstallPath(ResolvedPackageId.ToString(), ResolvedPackageVersion);
         var toolsPackagePath = Path.Combine(installPath, "tools");
-        Frameworks = _fileSystem.Directory.EnumerateDirectories(toolsPackagePath)
-            .Select(path => NuGetFramework.ParseFolder(Path.GetFileName(path))).ToList();
+
+        // Get available frameworks before deserializing tool configuration
+        // This allows us to provide better error messages if the tool requires a higher .NET version
+        List<NuGetFramework> availableFrameworks = [];
+        if (_fileSystem.Directory.Exists(toolsPackagePath))
+        {
+            availableFrameworks = _fileSystem.Directory.EnumerateDirectories(toolsPackagePath)
+                .Select(path => NuGetFramework.ParseFolder(Path.GetFileName(path))).ToList();
+        }
+        Frameworks = availableFrameworks;
```

Comment:
Instead of loading the available frameworks here and passing them to DeserializeToolConfiguration, load them in DeserializeToolConfiguration only if the tool settings aren't found.